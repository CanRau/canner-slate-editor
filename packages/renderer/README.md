

- [Renderers - @canner/slate-editor-renderer](./packages/slate-editor-renderer)
- [Serializer and deserializer - @canner/slate-editor-html](./packages/slate-editor-html)
