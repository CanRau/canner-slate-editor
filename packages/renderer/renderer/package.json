{
  "name": "@canner/slate-editor-renderer",
  "version": "1.3.0",
  "description": "Renderers for slate nodes and marks",
  "main": "lib/index.js",
  "scripts": {
    "test": "jest src",
    "prepublish": "../../../node_modules/.bin/babel src -d lib --copy-files"
  },
  "author": "",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "jest": "^22.4.2",
    "react-test-renderer": "^16.2.0"
  },
  "peerDependencies": {
    "react": "^0.14.0 || ^15.0.0 || 16.x",
    "react-dom": "^0.14.0 || ^15.0.0 || 16.x",
    "slate": "^0.33.0",
    "styled-components": "^3.2.1"
  },
  "dependencies": {
    "@canner/slate-helper-inline-adddata": "^0.1.2",
    "emoji-mart": "^2.5.0",
    "js-video-url-parser": "^0.2.7",
    "lodash.mapvalues": "^4.6.0",
    "quill-icons": "^0.6.0",
    "react-icons": "^2.2.7",
    "react-loading-image": "^0.4.2",
    "react-resizable": "^1.7.5",
    "slate-edit-table": "^0.15.0-alpha.4"
  }
}
