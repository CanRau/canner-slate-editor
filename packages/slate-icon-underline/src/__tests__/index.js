import React from "react";
import { Change, Value } from "slate";
import Underline from "../";
import renderer from "react-test-renderer";

test("create a Underline icon", () => {
  const initialValue = Value.fromJSON({
    document: {
      nodes: [
        {
          object: "block",
          type: "paragraph",
          nodes: [
            {
              object: "text",
              leaves: [
                {
                  text: "A line of text in a paragraph."
                }
              ]
            }
          ]
        }
      ]
    }
  });

  const component = renderer.create(
    <Underline change={new Change({ value: initialValue })} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
