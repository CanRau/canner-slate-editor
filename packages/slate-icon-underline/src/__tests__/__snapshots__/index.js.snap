// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a Underline icon 1`] = `
<span
  data-active={false}
  onClick={[Function]}
  onMouseDown={[Function]}
>
  <svg
    className={undefined}
    colorLabelClassName={undefined}
    colorStyle={undefined}
    evenClassName={undefined}
    height="18px"
    strokeMitterClassName={undefined}
    thinClassName={undefined}
    transparentClassName={undefined}
    viewBox="0 0 18 18"
    width="18px"
  >
    <path
      className="ql-stroke"
      d="M5,3V9a4.012,4.012,0,0,0,4,4H9a4.012,4.012,0,0,0,4-4V3"
    />
    <rect
      className="ql-fill"
      height="1"
      rx="0.5"
      ry="0.5"
      width="12"
      x="3"
      y="15"
    />
  </svg>
</span>
`;
